ctam/test_runner.py

```diff
@@ -908,7 +908,103 @@ def generate_domain_test_report(self):
+                                if isinstance(value, dict) and 'value' in value:
+                                    json_data[section_name][key] = value['value']
+                                elif not isinstance(value, dict):
+                                    json_data[section_name][key] = value
+
+                        elif file_name == 'package_info.json':
+                            section_name = "PACKAGE INFO"
+                            json_data[section_name] = {}
+                            for key, value in data.items():
+                                json_data[section_name][key] = value
+
+                        elif file_name == 'redfish_uri_config.json':
+                            section_name = "REDFISH URI CONFIG"
+                            json_data[section_name] = {}
+                            for key, value in data.items():
+                                if isinstance(value, dict):
+                                    json_data[section_name][key] = {k: (json.dumps(v) if isinstance(v, (list, dict)) and v not in (None, '', {}, []) else str(v)) for k, v in value.items() if v not in (None, '', {}, [])}
+                                elif value not in (None, '', {}, []):
+                                    json_data[section_name][key] = str(value)
+
+                        elif file_name == 'redfish_response_messages.json':
+                            section_name = "REDFISH RESPONSE MESSAGES"
+                            json_data[section_name] = {}
+                            for key, value in data.items():
+                                if value:
+                                    json_data[section_name][key] = value
+
+                        elif file_name == 'test_runner.json':
+                            section_name = "TEST RUNNER"
+                            json_data[section_name] = {}
+                            exclude_keys = ["$schema", "$id", "title", "description"]
+                            for key, value in data.items():
+                                if key not in exclude_keys:
+                                    # Skip keys with empty values
+                                    if value not in (None, '', [], {}):
+                                        if isinstance(value, list):
+                                            # Convert list to a single line (comma-separated)
+                                            json_data[section_name][key] = ', '.join(str(v) for v in value if v)
+                                        elif isinstance(value, dict):
+                                            for sub_key, sub_value in value.items():
+                                                json_data[section_name][f"{key}_{sub_key}"] = sub_value
+                                        else:
+                                            json_data[section_name][key] = value
+            
+            json_file_path = os.path.join(config_folder_path, "data.json")
```

Comment:
data.json file needs to be sanitized, depending on the dut_info.json.

Reply:
Ohh.....thanks for catching this. Will get back.